livre/lib/data/RawDataSource.h

```diff
@@ -0,0 +1,61 @@
+/* Copyright (c) 2011-2016, EPFL/Blue Brain Project
+ *                     Ahmet Bilgili <ahmet.bilgili@epfl.ch>
+ *
+ * This file is part of Livre <https://github.com/BlueBrain/Livre>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#ifndef _RawDataSource_h_
+#define _RawDataSource_h_
+
+#include <livre/core/data/DataSourcePlugin.h>
+
+#include <livre/lib/types.h>
+
+namespace livre
+{
+
+/**
+ * Reads a *.raw data with given details or nrrd volume. The limitation
+ * is, the volumes should fit into the GPU memory.
```

Comment:
I couldn't see a check for it, though, is this done somewhere else? What happens to me as a user if I open a huge file?